### PR TITLE
Integration Test Bugfixes

### PR DESCRIPTION
**Pull request by Langenbahn**
## Change Summary

Goal of this PR: Run a manual integration test of the prober on one machine using the newly-added metadata feature, fixing bugs as they arise

Changes:
Fixed an issue where logs had no payload because the structs in `probe/logger.go` did not have json descriptors
Fixed an issue with metadata where data was not in the foreseen format (Had one additional layer of escaping of newline characters and escaped double quotes)
Added a field in the config proto to specify the retry interval for registering with the server
Refactored logger to function in a more-object-oriented way, addressing the problem of a possible segmentation fault if an error was logged before metadata was collected
Provided default logging destination values for use before hostname/metadata are collected
Updated test cases/`fakeLogger` in probe to address above changes in the `Logger` interface
Improved comment coverage of exported functions and struct members

### CODE REVIEW

File: Controller/src/controller/rpc.go [hunk 1]
```diff
@@ -70,13 +70,13 @@ func makeCert() error {
 	if err != nil {
 		return err
 	}
-	config.Metadata.Cert = cert
+	*config.Metadata.Cert = string(cert)
```
Comment (brotherlogic2) [inline Controller/src/controller/rpc.go:73]: config.GetMetadta().Cert = proto.String(string(cert)) ?

File: Probe/src/probe/logger.go [hunk 2]
```diff
@@ -37,18 +37,18 @@ type CloudLogger struct {
 
 //TODO(langenbahn): add log/error destinations in config proto
 type probeLog struct {
-	sendTime  string
-	probeType string
-	latency   int
-	state     string
-	region    string
-	token     string
+	SendTime  string `json:"sendTime"`
```
Comment (brotherlogic2) [inline Probe/src/probe/logger.go:40]: nit: Capitalized struct members should have comment, e.g. //SendTime is...

File: Probe/src/probe/logger.go [hunk 3]
```diff
@@ -81,7 +81,7 @@ func (c *CloudLogger) LogFatalf(desc string, args ...interface{}) {
 // Log errors to specified log
 func (c *CloudLogger) LogError(desc string) {
 	//TODO(langenbahn) add any other useful information for errors
-	el := &errorLog{desc, probeConfigs.Probe[0].GetRegion(), deviceToken}
+	el := &errorLog{desc, hostname, deviceToken}
```
Comment (brotherlogic2) [inline Probe/src/probe/logger.go:84]: Where does hostname and deviceToken come from - should they be members of CloudLogger?

File: Probe/src/probe/rpc.go [hunk 4]
```diff
@@ -44,12 +46,20 @@ var (
...
-	items := strings.Split(raw, "\n")
+	items := strings.Split(raw, "commonInstanceMetadata.items")
 	var probeData []string
 	for i, item := range items {
 		// Search for item "probeData" key, manipulate the associated value at the next index
 		if strings.Contains(item, "probeData") {
+			// data will come in the form of: 'value: "DATA"'
 			probeData = strings.SplitN(items[i+1], ": ", 2)
+			// Remove trailing newline character from cert for unquoting
+			probeData[1] = probeData[1][:len(probeData[1]) - 1]
```
Comment (brotherlogic2) [inline Probe/src/probe/rpc.go:57]: strings.Trim() would be better for this.

Comment (Langenbahn) [reply Probe/src/probe/rpc.go]: addressed in 024773b

Comment (Langenbahn) [reply Probe/src/probe/rpc.go]: addressed in 024773b

File: Probe/src/probe/logger.go (hunk 2 above)
Comment (Langenbahn) [reply Probe/src/probe/logger.go]: addresesd in 024773b

File: Probe/src/probe/logger.go (hunk 3 above)
Comment (Langenbahn) [reply Probe/src/probe/logger.go]: addressed in ef6a24c

File: Controller/src/controller/rpc.go (hunk 1 above)
Comment (Langenbahn) [reply Controller/src/controller/rpc.go]: addressed in 024773b

**Pull request merged**